generator/src/main/java/com/algorand/sdkutils/listeners/GoGenerator.java

```diff
@@ -0,0 +1,678 @@
+package com.algorand.sdkutils.listeners;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.StringTokenizer;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import com.algorand.sdkutils.listeners.Publisher.Events;
+import com.algorand.sdkutils.utils.StructDef;
+import com.algorand.sdkutils.utils.Tools;
+import com.algorand.sdkutils.utils.TypeDef;
+
+public class GoGenerator extends Subscriber {
```

Comment:
I really think we need some sort of template system to generate the code. Once we build up all the models/paths/parameters we should be able to pipe the metadata into a series of templates which could be swapped out based on the SDK being generated.

We could either have a property file that maps "types" to SDK types, have a thin per-sdk mapping function, or use the template language to convert them in each template. I prefer keeping the SDK specific business logic in the template engine.

If we don't do something like that it seems like we're going to have 4 nearly identical implementations for this fairly large and complex file.

Reply:
Your observation is in place.
However, I worry that "4 nearly identical" may be problematic. I think individual customizations might be more difficult with templates than individual java programs.